core/dbt/tests/util.py

```diff
@@ -13,6 +14,35 @@
 from dbt.context import providers
 from unittest.mock import patch
 
+# =============================================================================
+# Test utilities
+#   run_dbt
+#   run_dbt_and_capture
+#   get_manifest
+#   copy_file
+#   rm_file
+#   write_file
+#   read_file
+#   get_artifact
+#   update_config_file
+#   get_unique_ids_in_results
+#   check_results_nodes_by_name
+#   check_result_nodes_by_unique_id
+
+# SQL related utilities
+#   run_sql_with_adapter
+#   relation_from_name
+#   check_relation_types (table/view)
+#   check_relations_equal
+#   check_relations_equal_with_relations
+#   check_table_does_exist
+#   check_table_does_not_exist
+#   get_relation_columns
+#   update_rows
+#      generate_update_clause
```

Comment:
I think 'get_artifact' should be okay to document. I'm not totally sure about 'get_manifest', mainly because it will load an internal version of the manifest that I'm not sure we want people to rely on. Most of the rest of the test utilities are pretty innocuous. I'm not totally clear on how useful they would be to users, because my imagination is that they'd be testing their own project, and most of those are helpful for modifying/updating a project.

in SQL related utilities, I think update_rows is still kind of funky, i.e. the interface is very specific to a couple of particular tests. And I'd like to look a bit more at run_sql_with_adapter (which is mostly designed to be used by the project.run_sql method). I think check_relations_equal, relation_from_name, check_relation_types, check_table_does_(not)_exist, and get_relation_columns are okay.  We will be expecting adapter contributors to be dealing with those utilities. I'm not totally clear how much uptake they'd get with general users.

Reply:
The principal intended user here is very much the adapter plugin maintainer, and I think that should inform how we publish/document the first cut of this interface. If it proves valuable for package/project maintainers as well, all the better.

- Makes sense that `get_manifest` wants to be private, given that it's largely the output of _parsing_, and ought not vary across adapters
- Agree that `update_rows` is not my favorite. I'd like to see [all these methods](https://github.com/dbt-labs/dbt-core/blob/0b92f04683cff108e3c7ad1d44572378b902ba03/core/dbt/adapters/base/impl.py#L1052-L1164) refactored to use SQL wrapped in Jinja macros, rather than in Python f-strings. We can make that change independently of the test utilities, though.